mypy_django_plugin/transformers/choices.py

```diff
@@ -0,0 +1,87 @@
+from mypy.nodes import MemberExpr, NameExpr, TypeInfo
+from mypy.plugin import AttributeContext
+from mypy.typeanal import make_optional_type
+from mypy.types import AnyType, Instance, TupleType, TypeOfAny, get_proper_type
+from mypy.types import Type as MypyType
+
+from mypy_django_plugin.lib import fullnames, helpers
+
+
+def transform_into_proper_attr_type(ctx: AttributeContext) -> MypyType:
+    """
+    A `get_attribute_hook` to make `.choices` and `.values` optional if `__empty__` is defined.
+
+    When the `__empty__` label is specified, the `choices` and `values` properties can return a
+    blank choice/value. This hook will amend the type returned by those properties.
+    """
+    if isinstance(ctx.context, MemberExpr):
+        method_name = ctx.context.name
+    else:
+        ctx.api.fail("Unable to resolve type of choices property", ctx.context)
+        return AnyType(TypeOfAny.from_error)
+
+    expr = ctx.context.expr
+
+    if isinstance(expr, MemberExpr):
+        expr = expr.expr
+
+    if isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo):
+        node = expr.node
+    else:
+        ctx.api.fail("Unable to resolve type of choices property", ctx.context)
+        return AnyType(TypeOfAny.from_error)
+
+    default_attr_type = get_proper_type(ctx.default_attr_type)
+
+    if not node.is_enum or not node.has_base(fullnames.CHOICES_CLASS_FULLNAME):
+        return default_attr_type
+
+    # Enums with more than one base will treat the first base as the mixed-in type.
+    base_type = node.bases[0] if len(node.bases) > 1 else None
+
+    # When `__empty__` is defined, the `.choices` and `.values` properties will include `None` for
+    # the blank choice which is labelled by the value of `__empty__`.
+    has_blank_choice = "__empty__" in node.names
```

Comment:
Should this respect `__empty__` defined in base classes? It would be safer to use 

```suggestion
    has_blank_choice = node.get("__empty__")
```

Reply:
I don't think so. When I implemented the choices stuff, it was only considered that `__empty__` would be defined on the class that introduced the members.

Reply:
> it's a pattern to be encouraged

Yeah, I undertand that. But, mypy is not about "encouriging", but about "it actually works". So, if this works in runtime, we should make it possible to check with mypy as well. 

Unless there are some big and complex problems.

Reply:
Ok. I'll look into it.

Reply:
Added this change 👍🏻